### DISCUSSION TITLE
Mix mp3 with mic input

### DISCUSSION THREAD

**Discussion by donnersm** [pschatzmann/arduino-audio-tools#1767] [Q&A] [answered]
Hi, Maybe someone can push me in the right direction. I looked at the examples but I don't really see what i need. 
Using an ESP32 and a I2S input and output, I want to mix an incoming input(mic) stream with an MP3 stream. So I will store mp3 files on SD and play them using the i2s output but before outputting the mp3 sound, it should be mixed with the input coming from mic.. like a karaoke machine.... on top of that would be great if i can manipulate the mic input....like with a vocoder... This would be used in a cosplay....like storm trooper uniform or robot... Anyone an idea what examples to look at or maybe even an idea how to mix those signals before output?

**Comment by pschatzmann** [comment, marked as answer]
There is a Wiki that covers Mixing!
I would try an input mixer and read from the microphone and from an EncodedAudioStream.

**Comment by donnersm** [comment]
Thanks, I found it
